### PR TITLE
Api to get resources from one or multiple catalogs

### PR DESCRIPTION
**Pull request by pratap0007**
This patch adds a catalogs params to query api and
which takes one or multiple catalogs name

endpoint: `/query?catalogs=<catalog_name>&catalogs=<catalog_name>`

Signed-off-by: Shiv Verma <shverma@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: api/v1/design/resource.go [hunk 1]
```diff
@@ -33,12 +33,15 @@ var _ = Service("resource", func() {
...
 
 	Method("Query", func() {
-		Description("Find resources by a combination of name, kind and tags")
+		Description("Find resources by a combination of name, kind,catalog and tags")
 		Payload(func() {
 			Attribute("name", String, "Name of resource", func() {
 				Default("")
 				Example("name", "buildah")
 			})
+			Attribute("catalogs", ArrayOf(String), "Catalogs of resource to filter by", func() {
```
Comment (SM43) [inline api/v1/design/resource.go:42]: you need do this changes for non `/v1` (preview) API also

File: api/pkg/shared/resource/resource.go [hunk 2]
```diff
@@ -314,6 +316,19 @@ func filterResourceName(match, name string) func(db *gorm.DB) *gorm.DB {
...
 
+func filterByCatalogs(catalogs []string) func(db *gorm.DB) *gorm.DB {
+	if len(catalogs) == 0 {
+		return noop
+	}
+
+	catalogs = lower(catalogs)
+	return func(db *gorm.DB) *gorm.DB {
+		return db.Model(&model.Resource{}).
+			Joins("JOIN catalogs as ct on ct.id = resources.catalog_id").
```
Comment (SM43) [inline api/pkg/shared/resource/resource.go:327]: I am not sure if we require `join` for this, we already `Preload` catalog data so we can just add a where statement. wdyt?

File: api/v1/design/resource.go (hunk 1 above)
Comment (SM43) [reply api/v1/design/resource.go:42]: i said non-preview `also`

**Comment by PuneetPunamiya** [comment]
@pratap0007  could you please rebase the pr once ??

File: api/pkg/shared/resource/resource_test.go [hunk 3]
```diff
@@ -142,6 +142,26 @@ func TestQuery_ByTags(t *testing.T) {
...
+		Limit: 100,
+	}
+
+	res, err := req.Query()
+	assert.NoError(t, err)
+
+	assert.Equal(t, 1, len(res))
+	assert.Equal(t,"catalog-community",res[0].Catalog.Name)
+}
+
```
Comment (PuneetPunamiya) [inline api/pkg/shared/resource/resource_test.go:164]: I think we should also add a negative test for this where we don't find any resource

**Comment by PuneetPunamiya** [comment]
Thanks :call_me_hand: 

/lgtm

**Pull request merged**